render/render.js

```diff
@@ -101,13 +101,25 @@ module.exports = function($window) {
 		return element
 	}
 	function createComponent(parent, vnode, hooks, ns, nextSibling) {
-		vnode.state = Object.create(vnode.tag)
-		var view = vnode.tag.view
-		if (view.reentrantLock != null) return $emptyFragment
-		view.reentrantLock = true
-		initLifecycle(vnode.tag, vnode, hooks)
-		vnode.instance = Vnode.normalize(view.call(vnode.state, vnode))
-		view.reentrantLock = null
+		var sentinel
+		if (typeof vnode.tag === "function") {
+			vnode.state = null
+			sentinel = vnode.tag
+			if (sentinel.$$reentrantLock$$ != null) return $emptyFragment
+			sentinel.$$reentrantLock$$ = true
```

Comment:
The amount fo `$` show how uneasy I was about adding a property on a user-provided class. Adding it on a function is a lesser worry because people rarely use functions as dictionaries, much less iterate over their keys or look for `$$propertiesThatLookLikeThis$$`... We may attach the lock to `vnode.tag.prototype.view` instead, but it would require duplicating the logic in the `typeof vnode.tag === "function"` block. It wouldn't be slower (as many total branches on a single initialization since the paths are mututally exclusive), but a bit bulkier (though not much, since there's exact duplication that gzips well).